Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ActionManager/Action/ActionManager.h>
+
+namespace AzToolsFramework
+{
+    ActionManager::ActionManager()
+    {
+        AZ::Interface<ActionManagerInterface>::Register(this);
+    }
+
+    ActionManager::~ActionManager()
+    {
+        AZ::Interface<ActionManagerInterface>::Unregister(this);
+
+        ClearActionContextMap();
+    }
+
+    ActionManagerOperationResult ActionManager::RegisterActionContext(
+        QWidget* widget,
+        AZStd::string_view identifier,
+        AZStd::string_view name,
+        AZStd::string_view parentIdentifier)
+    {
+        if (!widget)
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Action Manager - Could not register action context \"%.s\" to a null widget.", AZ_STRING_ARG(identifier))
+            );
+        }
+
+        m_actionContexts.insert(
+            {
+                identifier,
+                new EditorActionContext(identifier, name, parentIdentifier, widget)
+            }
+        );
+
+        return AZ::Success();
+    }
+
+    ActionManagerOperationResult ActionManager::RegisterAction(
+        AZStd::string_view contextIdentifier,
+        AZStd::string_view identifier,
+        AZStd::string_view name,
+        AZStd::string_view description,
+        AZStd::string_view category,
+        [[maybe_unused]] AZStd::string_view iconPath,
+        AZStd::function<void()> handler)
+    {
+        if (!m_actionContexts.contains(contextIdentifier))
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Action Manager - Could not register action \"%.s\" - context \"%.s\" has not been registered.",
+                AZ_STRING_ARG(identifier),
+                AZ_STRING_ARG(contextIdentifier)
+            ));
+        }
+
+        if (m_actions.contains(identifier))
+        {
+            return AZ::Failure(AZStd::string::format(
+                "Action Manager - Could not register action \"%.s\" twice.", AZ_STRING_ARG(identifier)));
+        }
+
+        m_actions.insert(
+            {
+                identifier,
+                EditorAction(m_actionContexts[contextIdentifier]->GetWidget(), identifier, name, description, category, handler)
+            }
+        );
+        return AZ::Success();
+    }
+
+    QAction* ActionManager::GetAction(AZStd::string_view actionIdentifier)
+    {
+        if (m_actions.contains(actionIdentifier))
+        {
+            return m_actions[actionIdentifier].GetAction();
+        }
+
+        return nullptr;
+    }
+
+    const QAction* ActionManager::GetActionConst(AZStd::string_view actionIdentifier)
```

Comment:
nit: This function can be annotated `const` as well

`const QAction* ActionManager::GetActionConst(AZStd::string_view actionIdentifier) const`

Reply:
I have actually tried this, but it doesn't build - operator [] on the map doesn't like it. I'll keep this in mind and try it out in follow-up PRs if it makes sense.